File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    arr = ["aya", "ye", "woo", "ma"]
    return babbling.reduce((acc,cur) => acc+isValid(cur,arr), 0)
}

const isValid = (word, arr, prevIdx = -1, stringIdx = 0) => {
    l1: while (stringIdx < word.length) {
        for (let i = 0; i<arr.length; i++) {
            if (i!==prevIdx && word.startsWith(arr[i], stringIdx)) {
                stringIdx += arr[i].length
                prevIdx = i
                continue l1
            }
        }
        return false
    }
    return true
}

File: 프로그래머스/unrated/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    return len([i for i in babbling if check(i)])

def check(s):
    prev = ''
    i=0
    while i < len(s):
        for c in ['aya', 'ye', 'woo', 'ma']:
            if s[i:].startswith(c) and prev != c :
                i += len(c)
                prev = c
                break
        else:
            return False
    return True